### ISSUE TITLE
`redundant-test-main-exit` bad `flag.Parse()` advice

### ISSUE THREAD

**Issue by scop** [mgechev/revive#1598]
**Describe the bug**

The version of revive included in golangci-lint 2.7.0 (I gather revive 1.13.0) flags use of `flag.Parse()` in tests in contexts where removing the call will lead to panic.
 
**To Reproduce**

Run golangci-lint 2.7.0 with revive & `enable-all-rules: true`.

Code:
```go
func TestMain(m *testing.M) {
	flag.Parse() // for .Short to work
	if testing.Short() {
		m.Run()
		return
	}
	// ...
```

Output:
```
path/to/some_test.go:25:2: redundant-test-main-exit: redundant call to flag.Parse in TestMain function, the test runner will handle it automatically as of Go 1.15 (revive)
```

Removing the `flag.Parse()` yields a panic (Go 1.25.5) when the tests are run (`go test ./...`):
```
panic: testing: Short called before Parse
```

**Expected behavior**
This use of `flag.Parse()` should not have been flagged.

**Logs**
-

**Desktop (please complete the following information):**
 - OS: [e.g. Ubuntu 18.04]: Ubuntu 25.10
 - Version of Go: 1.25.5

**Additional context**
-

**Comment by alexandear** [comment]
Thank you for reporting the issue. PR #1599 addresses it.
